sharktank/sharktank/ops/signatures.py

```diff
@@ -1341,13 +1341,29 @@ def _to_trampoline(d: SignatureDispatcher, tensor: AnyTensor, *args, **kwargs):
 
 
 @overridable
-def trace_tensor(key: str, *tensors: tuple[AnyTensor]):
+def trace_tensor(key: str, *tensors: tuple[AnyTensor, ...]):
+    """Trace tensor(s) in IREE runtime or in eager mode.
+
+    You can add trace_tensor into your model wherever you want. It will insert a
+    trace op into the IR. Then you can register a callback in the IREE runtime for
+    custom handling of the trace command during execution. For example recording the
+    tensor into a file. There is also a destination/sink for eager execution.
```

Comment:
Is there an example anywhere of how to do this?

Reply:
There are tests.